semver/parse_range.ts

```diff
@@ -404,9 +404,13 @@ function parseOperatorRanges(string: string): Comparator[] {
  * @returns A valid SemVer range
  */
 export function parseRange(range: string): Range {
-  return range
+  const result = range
     // remove spaces between operators and versions
-    .replaceAll(/(?<=<|>|=|~) +/g, "")
+    .replaceAll(/(?<=<|>|=|~|^)(\s+)/g, "")
```

Comment:
1. Addresses https://github.com/denoland/std/pull/5564#discussion_r1694506670
2. Required fix for this PR.

Reply:
Can you also add a test case for checking `^ 1.0.0` works?